stumpy/motifs.py

```diff
@@ -0,0 +1,412 @@
+        )
+
+        if len(query_matches) > min_neighbors:
+            top_k_indices.append(query_matches[:max_matches, 0])
+            top_k_values.append(query_matches[:max_matches, 1])
+
+        if len(query_matches) > 0:
+            for idx in query_matches[:, 0]:
+                core.apply_exclusion_zone(P, idx, excl_zone)
+        else:  # pragma: no cover
+            # I no match was found, still apply an exclusion zone around the candidate
+            # index, we do not want to consider it as a motif candidate again.
+            core.apply_exclusion_zone(P, candidate_idx, excl_zone)
+
+        candidate_idx = np.argmin(P[-1])
+
+    return top_k_indices, top_k_values
+
+
+def motifs(
+    T,
+    P,
+    excl_zone=None,
+    min_neighbors=1,
+    max_distance=None,
+    cutoff=None,
+    max_matches=10,
+    max_motifs=1,
+    normalize=True,
+):
+    """Find the top k motifs of the time series `T`.
+
+    A subsequence `Q` is considered a motif if there are at least `min_neighbor` other
+    matches in `T` (outside the exclusion zone) with distance less or equal to
+    `atol + rtol * profile_value`, where `profile_value` is the matrix profile
+    value of `Q`.
+
+    Parameters
+    ----------
+    T : ndarray
+        The time series of interest
+
+    P : ndarray
+        Matrix Profile of `T` (result of a self-join)
+
+    M_T : ndarray
+        Sliding mean of time series, `T`
+
+    Σ_T : ndarray
+        Sliding standard deviation of time series, `T`
+
+    excl_zone : int, default None
+        Size of the exclusion zone.
+        If `None`, defaults to `m/4`, where `m` is the length of `Q`.
+
+    min_neighbors : int, default 1
+        The minimum amount of similar matches a subsequence needs to have
+        to be considered a motif.
+        Defaults to `1`. This means, that a subsequence has to have
+        at least one similar match to be considered a motif.
+
+    max_distance : float or function, default None
+        Maximum distance between a motif candidate `Q` and a subsequence `S` and  for
+        `S` to be considered a match of the motif represented by `Q`.
+        If a function, then it has to be a function of one argument `D`, which will be
+        the distance profile of `Q` with `T` (a 1D numpy array of size `n-m+1`).
+        If None, defaults to `max(np.mean(D) - 2 * np.std(D), np.min(D))`, meaning
+        that least the closest match will be returned.
+
+    cutoff : float, default None
+        The largest matrix profile value a motif representative is allowed to have.
+        If `None`, defaults to `max(np.mean(P) - 2 * np.std(P), np.min(P))`
+
+    max_matches : int, default 10
+        The maximum amount of similar matches to be returned. The resulting
+        matches are sorted by distance, so a value of `10` means that the
+        indices of the most similar `10` subsequences is returned. If `None`,
+        all matches in the given tolerance range are returned.
```

Comment:
actually "all matches within `max_distance` of the motif representative will be returned.", since the max_matches refers to the number of matches of the motif representative. I clarified that.

Reply:
Oops! You're absolutely right. Thank you